third_party_libraries/kml.js

```diff
@@ -660,6 +399,86 @@ L.Util.extend(L.KML, {
 		return coords;
 	},
 
+
+	/**
+	 * Gets the style options for the placemarks leaflet marker
+	 *
+	 * @param {xml Element} place A placemark XML element
+	 * @param {path} style Style options for the placemark
+	 * @return {path[]} options The style options for the placemark
+	 */
+	getOptions: function (place, style) {
+		var i, j, el, options = {};
+		el = place.getElementsByTagName('styleUrl');
+
+		for (i = 0; i < el.length; i++) {
+			var url = el[i].childNodes[0].nodeValue;
+			for (var a in style[url]) {
+				options[a] = style[url][a];
+			}
+		}
+
+		return options;
+	},
+
+	/**
+	 * Creates a base layergroup element for the placemarks leaflet icon
+	 *
+	 * @param {xml Element} place A placemark XML element
+	 * @param {path[]} options The style options for the placemark
+	 * @param {xml} xml Xml data related to the placemark
+	 * @return {FeatureGroup} layer Basic location and style data pertaining to
+	 * 		the placemark
+	 */
+	initializeLayer: function (place, options, xml) {
+		var layers = [];
+		var parse = ['LineString', 'Polygon', 'Point'];
+		for (j in parse) {
+			var tag = parse[j];
+			el = place.getElementsByTagName(tag);
+
+			for (i = 0; i < el.length; i++) {
+				switch(tag) {
+					case 'LineString':
+  						var layer = this.parseLineString(el[i], xml, options);
+    						if (l) { layers.push(l); }
+    						break;
+     					case 'Polygon':
+						var l = this.parsePolygon(el[i], xml, options);
+						if (l) { layers.push(l); }
+						break;
+         				case 'Point':
+						var l = this.parsePoint(el[i], xml, options);
+						if (l) { layers.push(l); }
+						break;
+				}
+			}
+		}
+
+		if (!layers.length) {
+			return null;
+		}
+		var layer = layers[0];
+		if (layers.length > 1) {
+			layer = new L.FeatureGroup(layers);
+		}
+
+		return layer;
+	},
+
+
+		/**
+		 * Replaces open/close tags with safe strings
+		 * @param {string} String to sanitize
+		 * @return {string} Sanitized string
+		 */
+		 sanitizeHtml: function (string) {
```

Comment:
Sorry for nitpicking indentation, but this method and its docblock are indented one level too far. Everything else looks good though so poke that and I'm happy.